crates/dev-geth/src/main.rs

```diff
@@ -0,0 +1,105 @@
+//! `dev-geth` is a small program to help with running [geth] in dev mode to
+//! serve as an environment for automated tests. Geth doesn't provide a way to
+//! reset the dev environment, other than restarting the geth program itself.
+//! `dev-geth` exposes an HTTP port to which a POST request can be made in order
+//! to restart geth and provide a clean testing environment. Note that
+//! restarting the program will wait until geth starts responding to RPC
+//! requests before returning 200 to the caller.
+//!
+//! geth is the only node with dev mode which implements the
+//! `eth_createAccessList` endpoint. When hardhat adds support for
+//! `eth_createAccessList`, this can be removed from our infrastructure.
+//!
+//! [geth]: https://geth.ethereum.org/
```

Comment:
Not suggesting that we change it here - but what would be really cool (and help us avoid having needing to run integration tests with `--threads 1`) would be to start multiple `geth` instances bound to different ports for each test and have this helper manage them.

Reply:
Kind of related: This doesn't need to be separately run in docker. It would be better to make this a normal struct that is instantiated by the test that needs geth. The only reason to make it a separate container is so that the same geth can be used in multiple tests, but if each test is restarting it anyway, or with Nic's suggestion has it's own node, then this should be integrated directly into the test.